bgpd/bgp_addpath.c

```diff
@@ -432,6 +433,15 @@ void bgp_addpath_update_ids(struct bgp *bgp, struct bgp_node *bn, afi_t afi,
 	struct bgp_path_info *pi;
 	struct id_alloc_pool **pool_ptr;
 
+	/* Since we are keeping labeled-unicast routes in unicast table
+	 * addpath IDs are not generated because peercount is always
+	 * zero for SAFI_UNICAST.
+	 * And the result is that only the best path is sent to peers
+	 * instead of multiple paths.
+	 */
+	if (CHECK_FLAG(bn->flags, BGP_NODE_LABEL_CHANGED))
```

Comment:
I'm not super familiar with the MPLS implementation, so forgive me if this is a stupid question. Is this durable for all causes of best-path re-run? My thinking is, if this is happening in the SAFI_UNICAST table, could we have standard unicast paths mixed with the labeled unicast routes on this bgp_node? If that was the case, this function could get called without this flag set, which would be very dangerous.

Reply:
I cannot answer your question. It would be great for someone to answer this question familiar with MPLS...

Reply:
Are you concerned that in a table with mixed tables, we'll end up with the safi set to labeled rather than unicast because of this check?

Reply:
Yes. I think the AFI/SAFI used here should always correspond to the table in which the routes are stored. If we let this run with different AFI/SAFIs against the same bgp_node, we could allocate an addpath ID from one AFI/SAFI's allocator, and free it back to another, which could trigger asserts or bugs in the allocator (or let a prefix pick duplicate IDs.)

I'm thinking the way to fix this is by having the peer counts reflect the quantity of addpath users on a particular table in this case. In particular, bgp_addpath_set_peer_type would need to understand when adding/dropping an anycast peer in the labeled-unicast family, this is actually an anycast user in the standard anycast table.

Are there any other instances of multiple address families using the same table?

Reply:
>Are there any other instances of multiple address families using the same table?
AFAIK no.

Reply:
the labeled unicast safi is redirected into the unicast safi in FRR.

Reply:
which means it should work correctly, I think.